Multiple changes, including adding travis and github keys; new README; and updated tests

andrie opened the pull request:
Add new function `add_travis_user()`; imports `httr` to read travis key

gaborcsardi commented on R/users.R:
```diff
@@ -20,13 +20,13 @@
   assert_that(is_email_address(email))
   vault <- find_vault(vault)
-
+  
```
You have whitespace errors.

gaborcsardi commented on R/users.R:
```diff
@@ -20,13 +20,13 @@
          "To update it, remove the old key, and add the new one.")
   }
-
+  
```
Another whitespace error.

gaborcsardi commented on R/users.R:
```diff
@@ -45,6 +45,29 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
+#' Add a user via their Travis repo
+#' 
+#' @param travis_repo Name of Travis repository, usually in a format 
+#' <<username>>/<<repo>>
```
Best to make this code, in backticks: `` `<<username>>/<<repo>>` ``

gaborcsardi commented on R/users.R:
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
   assert_that(is_email_address(email))
   vault <- find_vault(vault)
-
+  
```
Whitespace error.

gaborcsardi commented on R/users.R:
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
     stop("User ", sQuote(email), " does not exist")
   }
-
+  
```
Whitespace error.

gaborcsardi commented on R/users.R:
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
   ## Get all secrets they have access to
   secrets <- list_user_secrets(vault, email)
-
+  
```
Whitespace.

gaborcsardi commented on R/users.R:
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
   ## Remove everything in one go. This is still not atomic, of course...
   file.remove(user_file, secrets)
-
+  
```
Whitespace.

gaborcsardi commented:
Thanks! Added some nitpicking comments. 

I'll probably also rewrite it to use `curl`, that's a bit lighter than `httr`, and we don't really need `httr` functionality here.

andrie commented:
Nitpicking is good.  I'm not entirely sure what you mean by "whitespace error" - I presume you don't want the empty lines.  I have deleted those.

I agree with taking the lighter weight dependency on `curl` instead of `httr`.  Will implement.

andrie commented:
Now imports from curl and jsonlite, instead of httr.
Also fixes an issue that when creating a package vault, the path is `./inst/vault`, but reading from an installed package, the path is `./vault`

andrie commented:
Added functionality for `add_github_user()`

andrie commented:
This last pull request adds a small bug fix, as well as new tests for `add_travis_user()` and `add_github_user()`.

I also renamed the keys in `./inst/user_keys` to `alice.pub`, `bob.pub`, etc. so the code is easier to read and the concepts easier to explain in a vignette and readme.

andrie commented:
I rewrote the `README` using `rmarkdown`, i.e. now there is a `README.Rmd` file

andrie commented:
I have refactored the tests into multiple files, and added additional tests for `update_secret()`, `share_secret()` and `local_key()`.  I also added code coverage into `.travis.yml` and added the code coverage badge.  Code coverage now is greater than 95%.

gaborcsardi commented:
Great, thanks!
